test/accel/TrackingManagerIntegration.test.cc

```diff
@@ -411,8 +417,12 @@ TEST_F(LarSphereOptical, run)
     {
         GTEST_SKIP() << "Skipping remaining tests since we've already failed";
     }
-    CELER_LOG(status) << "Run one more event";
-    rm.BeamOn(2);
+    // check if geometry is vecgeom surface model and skip if so
+    if (!using_surface_vg)
+    {
+        CELER_LOG(status) << "Beam on (second run)";
+        rm.BeamOn(2);
+    }
```

Comment:
I literally gave you the code to use:
```suggestion
    if (using_surface_vg)
    {
        GTEST_SKIP() << "VecGeom surface does not support multiple runs";
    }

    CELER_LOG(status) << "Beam on (second run)";
    rm.BeamOn(2);
```

This changes less code, makes the exceptional case the exception rather than the main path flow, and prints an informative message about why we're skipping the rest of the test. Please change in the other test as well.